lib/abilities.ex

```diff
@@ -9,6 +9,12 @@ defimpl Canada.Can, for: Pairmotron.User do
   def can?(%User{id: user_id}, action, %Group{owner_id: user_id})
     when action in [:edit, :update, :delete], do: true
 
+  def can?(user = %User{}, :show, group = %Group{}) do
+    group = group |> Pairmotron.Repo.preload(:users)
+    group.users
+      |> Enum.any?(fn guser -> guser.id == user.id end)
```

Comment:
> Could lines 14-15 be simplified to `Enum.member?(group.users, user)`

No unfortunately. That was the cause of the last bug I found. The user and the group.users have to match exactly, including preloaded associations and other things, for `Enum.member?` to work.

> I think that this permission check should be in the GroupPairController show action itself :(.

I leaned that way initially too. It fit perfectly as you say, but could be the cause of some trouble later. Want me to remove this now?